internal/service/advancedcluster/data_source_advanced_cluster.go

```diff
@@ -235,50 +250,119 @@ func DataSource() *schema.Resource {
 
 func dataSourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
 	connV2 := meta.(*config.MongoDBClient).AtlasV2
+	connLatest := meta.(*config.MongoDBClient).AtlasV2Preview
+
 	projectID := d.Get("project_id").(string)
 	clusterName := d.Get("name").(string)
+	useReplicationSpecPerShard := false
+	var replicationSpecs []map[string]any
+	var clusterID string
 
-	cluster, resp, err := connV2.ClustersApi.GetCluster(ctx, projectID, clusterName).Execute()
-	if err != nil {
-		if resp != nil && resp.StatusCode == http.StatusNotFound {
-			return nil
+	if v, ok := d.GetOk("use_replication_spec_per_shard"); ok {
+		useReplicationSpecPerShard = v.(bool)
+	}
+
+	if !useReplicationSpecPerShard {
+		clusterDescOld, resp, err := connV2.ClustersApi.GetCluster(ctx, projectID, clusterName).Execute()
+		if err != nil {
+			if resp != nil {
+				if resp.StatusCode == http.StatusNotFound {
+					return nil
+				}
+				if err.Error() == "ASYMMETRIC_SHARD_UNSUPPORTED" {
+					return diag.FromErr(fmt.Errorf("please add `use_replication_spec_per_shard = true` to your data source configuration to enable asymmetric shard support. Refer to documentation for more details. %s", err))
+				}
+			}
+			return diag.FromErr(fmt.Errorf(errorRead, clusterName, err))
+		}
+
+		clusterID = clusterDescOld.GetId()
+
+		if err := d.Set("disk_size_gb", clusterDescOld.GetDiskSizeGB()); err != nil {
+			return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "disk_size_gb", clusterName, err))
+		}
+
+		replicationSpecs, err = FlattenAdvancedReplicationSpecsOldSDK(ctx, clusterDescOld.GetReplicationSpecs(), d.Get("replication_specs").([]any), d, connLatest)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "replication_specs", clusterName, err))
+		}
+
+		diags := setCommonSchemaFields(d, convertClusterDescToLatestExcludeRepSpecs(clusterDescOld))
+		if diags.HasError() {
+			return diags
+		}
+	} else {
+		clusterDescLatest, resp, err := connLatest.ClustersApi.GetCluster(ctx, projectID, clusterName).Execute()
+		if err != nil {
+			if resp != nil && resp.StatusCode == http.StatusNotFound {
+				return nil
+			}
+			return diag.FromErr(fmt.Errorf(errorRead, clusterName, err))
+		}
+
+		clusterID = clusterDescLatest.GetId()
+
+		replicationSpecs, err = flattenAdvancedReplicationSpecsDS(ctx, clusterDescLatest.GetReplicationSpecs(), d, connLatest)
+		if err != nil {
+			return diag.FromErr(fmt.Errorf(ErrorClusterAdvancedSetting, "replication_specs", clusterName, err))
+		}
```

Comment:
within this path, would we need to also disk_size_gb by setting the value defined in the nested hardware spec object?

Reply:
you mean set the root level disk_size_gb based on the nested value?

Reply:
yes was thinking this case

Reply:
discussed offline, will keep as is for now